### PR TITLE
[WIP] Support for modals

### PR DESCRIPTION
**Pull request by dignati**
This adds support for modals. I opened the pull request to discuss how the modal should be activated/faded in. Some sort of channel or function to call?

### CODE REVIEW

**Comment by sritchie** [comment]
I think react-bootstrap did this with a mixin, and modified the classes of child elements to force CSS transitions. If we can match the API those guys use, where if you tweak the input parameter it triggers the animation, I think we'll be good here. Take a look at that, let me know what you think, and I'll be happy to dive in with more feedback and reviews. Both @dpetrovics and I are around, so we'll be able to respond pretty fast to any questions. Thanks so much for this!!

**Comment by dignati** [comment]
Oh I'm happy to contribute to this great project!

As I said in #7, implementing the FadeMixin feels very hacky (timeouts, string replacement to add and remove classes etc.). Would it be an option to handle the fade logic inside the modal component and provide the user a function to open the modal? I understand that this is not how react-bootstrap did it but maybe this approach could actually be better.

**Comment by dpetrovics** [comment]
Definitely-- we're not tied to mimicking their approach. If you think you
can cleanly implement it by just handling the logic within the modal
component that's fine by me. Thanks for this!

**Comment by dignati** [comment]
Am I missing something or is it quite easy now? The user build the modal component, providing the ```:visible?``` key. Depending on that, we add the class ```in``` or not, forcing the animation.

File: src/om_bootstrap/modal.cljs [hunk 1]
```diff
@@ -1,104 +1,54 @@
...
+             (d/div {:class "modal-dialog"}
+                    (d/div {:class "modal-content"}
+                           (d/div {:class "modal-header"}
+                                  (when (:close-button? bs)
+                                    (d/button {:type         "button"
+                                               :class        "close"
+                                               :aria-hidden  true
+                                               :on-click (fn [_] (hide! owner))}
+                                              "×"))
+                                  (u/clone-with-props (:header bs) {:class "modal-title"}))
```
Comment (dignati) [inline src/om_bootstrap/modal.cljs]: Should we leave the actual header element to the user or should we wrap it in a ```h4``` by default?

Comment (sritchie) [inline src/om_bootstrap/modal.cljs]: I'd say leave it up to the user - they can control the styling on their own with that class, or they can do their own wrapping and pass stuff in.

**Comment by sritchie** [comment]
Hey @dignati, we'll be around today, then back in the office on Wednesday, so apologies in advance for a little radio silence. Looking really good here.

**Comment by dignati** [comment]
Okay that should cover everything except the contained modal with backdrop rendering. Maybe add that in a later pull-request?

**Comment by dignati** [comment]
Sorry for having merge conflicts btw, is there any way I can fix that for you?

**Comment by sritchie** [comment]
Best thing to do is to merge our develop branch back into yours, fix merge conflicts, then push again. Then we should be able to merge in.

**Comment by dignati** [comment]
@sritchie Alright, should be ready to merge now if the test passes.

**Comment by sritchie** [comment]
Nice! We'll merge when tests pass.

**Comment by dignati** [comment]
Great, I'm really excited to get this merged. This was the last missing piece to use om-bootstrap in production!

**Comment by sritchie** [comment]
Okay, released as 0.3.5. Definitely send us a link once you get it into production!

**Comment by julienba** [comment]
The sample is not working for me with the 0.3.5 (nothing appear and no error in the chrome console).
Nothing appear on http://om-bootstrap.herokuapp.com/components#modals too

Did I missing something ?

**Comment by sritchie** [comment]
@dignati, can you take a look? Thanks for the note, @julienba. We'll get this fixed up.

**Pull request merged**